### PR TITLE
Documentation: Improve Testing Overview page

### LINKED ISSUE #3838: Testing Overview docs formatting
Ref: https://wordpress.org/gutenberg/handbook/reference/testing-overview/

In the first section, the shell scripts aren't highlighting correctly - code blocks don't work in lists, for some reason. I'm inclined to move the comment to be a sentence in the first list item, then the code can just be the command.

There are a bunch of soft line breaks in document. While these are allowed by the Markdown spec, Jetpack Markdown turns them into `<br/>`s, to match WordPress' behaviour.

See #3818.

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This PR fixes #3838 by updating the existing Testing Overview docs formatting:

> In the first section, the shell scripts aren't highlighting correctly - code blocks don't work in lists, for some reason. I'm inclined to move the comment to be a sentence in the first list item, then the code can just be the command.
> There are a bunch of soft line breaks in document. While these are allowed by the Markdown spec, Jetpack Markdown turns them into <br/>s, to match WordPress' behaviour.

It also follows @youknowriad suggestion https://github.com/WordPress/gutenberg/pull/3818#issuecomment-349639474:

> This looks good to me, I think we should also move our current Testing docs (in the CONTRIBUTING.md) file to this newly created doc for consistency.

## Types of changes

Docs changes only.

### CODE REVIEW

File: docs/testing-overview.md [hunk 1]
```diff
@@ -151,12 +148,44 @@ test( 'should contain mars if planets is true', () => {
...
+It's tempting to snapshot deep renders, but that makes for huge snapshots. Additionally, deep renders no longer test a single component, but an entire tree. With `shallow`, we snapshot just the components that are directly rendered by the component we want to test.
+
+### Code Coverage
+
+Code coverage is measured for each PR using the [codecov.io](https://codecov.io/gh/WordPress/gutenberg) tool.
+[Code coverage](https://en.wikipedia.org/wiki/Code_coverage) is a way of measuring the amount of code covered by the tests in the test suite of a project.  In Gutenberg, it is currently measured for JavaScript code only.
+
+## End to end Testing
+
+If you're using the built-in local environment above, you can run the e2e tests locally using this command:
```
Comment (youknowriad) [inline docs/testing-overview.md]: The local environment is no longer "above", we should add a link :)

**Review by youknowriad — APPROVED** [review]
Aside the minor comment, this looks good 👍

File: docs/testing-overview.md (hunk 1 above)
Comment (gziolo) [reply docs/testing-overview.md]: Good catch. I will update :)

File: docs/testing-overview.md [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
+
+Tests for JavaScript use [Jest](http://facebook.github.io/jest/) as the test runner. If needed, you can also use [Enzyme](https://github.com/airbnb/enzyme) for React component testing.
+
+Assuming you've followed the instructions above to install Node and project dependencies, tests can be run from the command-line with NPM:
+
+```
+npm test
+```
+
+To run unit tests only, use `npm run test-unit` instead.
```
Comment (BE-Webdesign) [inline docs/testing-overview.md]: Should we make it more clear that this bypasses the linter?

To run unit tests only, without the linter, use `npm run test-unit` instead.

**Review by BE-Webdesign — APPROVED** [review]
Same sentiment as youknowriad.

**Review by pento — APPROVED** [review]
Looks good to me!

File: docs/testing-overview.md (hunk 2 above)
Comment (gziolo) [reply docs/testing-overview.md]: 👍

**Pull request merged**